### PR TITLE
Fix/carousel android rtl jumps

### PR DESCRIPTION
**Pull request by NitzanWix**
## Description
There was a problem with the currentPage calculation. I removed the usage of the getCalcIndex from the goToPage and moved the page calculation in the goToNextPage. On android RTL the children is passed reversed for some reason so I revered so the first item will show first.

## Changelog
Carousel - Fixed Android RTL jumps on random items.

### CODE REVIEW

File: src/components/carousel/index.tsx [hunk 1]
```diff
@@ -501,11 +502,7 @@ class Carousel extends Component<CarouselProps, CarouselState> {
     const ScrollContainer = animatedScrollOffset ? Animated.ScrollView : ScrollView;
     const contentOffset = this.getInitialContentOffset(snapToOffsets);
     return (
-      <View
-        animated={animated}
-        style={[{marginBottom}, containerStyle]}
-        onLayout={this.onContainerLayout}
-      >
+      <View animated={animated} style={[{marginBottom}, containerStyle]} onLayout={this.onContainerLayout}>
```
Comment (M-i-k-e-l) [inline src/components/carousel/index.tsx:505]: Is this a change or a prettify?
If it's the latter, please do it in a separate PR or at least a separate commit (preferably the first or the last).

File: src/components/carousel/index.tsx [hunk 2]
```diff
@@ -455,23 +456,23 @@ class Carousel extends Component<CarouselProps, CarouselState> {
...
         </View>
       );
     }
   }
 
   renderAccessibleLayout() {
-    const {containerStyle, children, testID} = this.props;
-
+    const {containerStyle, children: propsChildren, testID} = this.props;
+    const children = Constants.isRTL && Constants.isAndroid ? React.Children.toArray(propsChildren) : propsChildren;
```
Comment (M-i-k-e-l) [inline src/components/carousel/index.tsx:475]: What is this for?

File: src/components/carousel/index.tsx [hunk 3]
```diff
@@ -239,10 +240,8 @@ class Carousel extends Component<CarouselProps, CarouselState> {
     if (containerWidth) {
       const spacings = pageWidth === containerWidth ? 0 : this.getItemSpacings(this.props);
       const initialBreak = pageWidth - (containerWidth - pageWidth - spacings) / 2;
-      const snapToOffsets = _.times(
-        presenter.getChildrenLength(this.props),
-        index => initialBreak + index * pageWidth + this.getContainerMarginHorizontal()
-      );
+      const snapToOffsets = _.times(presenter.getChildrenLength(this.props),
```
Comment (M-i-k-e-l) [inline src/components/carousel/index.tsx:243]: Is this a change or a prettify?
If it's the latter, please do it in a separate PR or at least a separate commit (preferably the first or the last).

File: src/components/carousel/index.tsx [hunk 4]
```diff
@@ -195,28 +195,29 @@ class Carousel extends Component<CarouselProps, CarouselState> {
...
   }
 
   goToNextPage() {
     const {currentPage} = this.state;
     const pagesCount = presenter.getChildrenLength(this.props);
     const {loop} = this.props;
     let nextPageIndex;
-
+    if (currentPage === pagesCount + 1) {
+      return this.goToPage(0, false);
```
Comment (M-i-k-e-l) [inline src/components/carousel/index.tsx:207]: Did you mean to:
```
this.goToPage(0, false);
return;
```

File: src/components/carousel/index.tsx [hunk 5]
```diff
@@ -355,10 +354,11 @@ class Carousel extends Component<CarouselProps, CarouselState> {
   };
 
   onScrollEvent = Animated.event([
-    {nativeEvent: 
-      {contentOffset: 
-        // @ts-ignore
-        {y: this.props?.animatedScrollOffset?.y, x: this.props?.animatedScrollOffset?.x}
+    {
```
[the remark this answers is not in the document]
Comment (NitzanWix) [reply src/components/carousel/index.tsx:357]: Prettify. I create two separate commits. One for formattings and one for actual logic change.

File: src/components/carousel/index.tsx (hunk 2 above)
Comment (NitzanWix) [reply src/components/carousel/index.tsx:475]: For some on RTL android sends the children reversed.

File: src/components/carousel/index.tsx (hunk 3 above)
Comment (NitzanWix) [reply src/components/carousel/index.tsx:243]: Prettify. I created two separate commits. One for formattings and one for actual logic change.

File: src/components/carousel/index.tsx (hunk 4 above)
Comment (NitzanWix) [reply src/components/carousel/index.tsx]: Yes. Fixed

File: src/components/carousel/index.tsx [hunk 6]
```diff
@@ -454,14 +457,14 @@ class Carousel extends Component<CarouselProps, CarouselState> {
 
   renderCounter() {
     const {pageWidth, showCounter, counterTextStyle} = this.props;
-    const {currentStandingPage} = this.state;
```
Comment (M-i-k-e-l) [inline src/components/carousel/index.tsx:458]: Is this change needed?

**Review by M-i-k-e-l — COMMENTED** [review]
Looks like Android in RTL should move in the opposite direction

File: src/components/carousel/index.tsx [hunk 7]
```diff
@@ -492,20 +493,17 @@ class Carousel extends Component<CarouselProps, CarouselState> {
...
-    const {containerStyle, animated, horizontal, animatedScrollOffset, ...others} = this.props;
+    const {containerStyle, animated, horizontal, animatedScrollOffset, style, ...others} = this.props;
     const scrollContainerStyle = this.shouldUsePageWidth()
       ? {paddingRight: this.getItemSpacings(this.props)}
       : undefined;
     const snapToOffsets = this.getSnapToOffsets();
     const marginBottom = Math.max(0, this.getContainerPaddingVertical() - 16);
     const ScrollContainer = animatedScrollOffset ? Animated.ScrollView : ScrollView;
     const contentOffset = this.getInitialContentOffset(snapToOffsets);
+    const _style = Constants.isRTL && Constants.isAndroid ? [styles.invertedView, style] : style;
```
Comment (NitzanWix) [inline src/components/carousel/index.tsx:504]: I inverted the view on android rtl so it would behave the same as on ios.

File: src/components/carousel/index.tsx [hunk 8]
```diff
@@ -305,7 +307,7 @@ class Carousel extends Component<CarouselProps, CarouselState> {
   onMomentumScrollEnd = () => {
     // finished full page scroll
     const {currentStandingPage, currentPage} = this.state;
-    const index = this.getCalcIndex(currentPage);
+    const index = currentPage;
```
Comment (M-i-k-e-l) [inline src/components/carousel/index.tsx:310]: Why not remove `index`?

File: src/components/carousel/index.tsx [hunk 9]
```diff
@@ -446,7 +449,7 @@ class Carousel extends Component<CarouselProps, CarouselState> {
           color={color}
           {...others}
           numOfPages={pagesCount}
-          currentPage={this.getCalcIndex(this.state.currentPage)}
+          currentPage={this.state.currentStandingPage}
```
Comment (M-i-k-e-l) [inline src/components/carousel/index.tsx:452]: I'de destruct at the top of the function

**Review by M-i-k-e-l — APPROVED** [review]
Android looks good!
A couple of small notes and I did not test iOS (cannot install it ATM)

File: src/components/carousel/index.tsx (hunk 8 above)
Comment (NitzanWix) [reply src/components/carousel/index.tsx]: Yeah. Fixed
